File: Game-AdminPanel-Project/React-Frontend/admin-panel-frontend/src/services/userService.js
import axios from 'axios';

const API_URL = 'http://localhost:5113/api'; 

// Kullanıcı kaydını gerçekleştirir
const register = async (userData) => {
    try {
        console.log('Kayıt işlemi gerçekleştiriliyor:', userData);
        const response = await axios.post(`${API_URL}/user/register`, userData);
        console.log('API yanıtı:', response.data);
        return response.data;
    } catch (error) {
        if (error.response) {
            console.error('API kayıt hatası:', error.response.data);
            throw new Error(error.response.data.title || 'Kullanıcı adı veya e-posta zaten kullanılıyor');
        } else if (error.request) {
            console.error('API kayıt isteği hatası:', error.request);
            throw new Error('Sunucu yanıt vermedi');
        } else {
            console.error('API kayıt bilinmeyen hatası:', error.message);
            throw new Error('Bilinmeyen bir hata oluştu');
        }
    }
};

// Kullanıcı girişini gerçekleştirir
const login = async (username, password) => {
    try {
        console.log('Giriş işlemi gerçekleştiriliyor:', { username, password });
        const response = await axios.post(`${API_URL}/user/login`, {
            username,
            password
        });
        console.log('API yanıtı:', response.data);
        return response.data;
    } catch (error) {
        if (error.response) {
            console.error('API giriş hatası:', error.response.data);
            throw new Error(error.response.data.title || 'Kullanıcı adı veya şifre hatalı');
        } else if (error.request) {
            console.error('API giriş isteği hatası:', error.request);
            throw new Error('Sunucu yanıt vermedi');
        } else {
            console.error('API giriş bilinmeyen hatası:', error.message);
            throw new Error('Bilinmeyen bir hata oluştu');
        }
    }
};

const userService = {
    register,
    login
};

export default userService;

File: Game-AdminPanel-Project/ASP-Backend/AdminPanelBackend/Services/UserService.cs
using System.Threading.Tasks;
using Microsoft.Extensions.Logging;
using Microsoft.EntityFrameworkCore;
using AdminPanelBackend.Data;
using AdminPanelBackend.Models;

namespace AdminPanelBackend.Services
{
    public class UserService : IUserService
    {
        private readonly ApplicationDbContext _context;
        private readonly ILogger<UserService> _logger;

        public UserService(ApplicationDbContext context, ILogger<UserService> logger)
        {
            _context = context;
            _logger = logger;
        }

        public async Task<bool> Register(RegisterModel registerModel)
        {
            try
            {
                var existingUser = await _context.Users
                    .FirstOrDefaultAsync(u => u.Username == registerModel.Username || u.EmailAddress == registerModel.EmailAddress);

                if (existingUser != null)
                {
                    _logger.LogWarning("Kullanıcı adı veya e-posta zaten kullanılıyor.");
                    throw new InvalidOperationException("Kullanıcı adı veya e-posta zaten kullanılıyor.");
                }

                var user = new User
                {
                    Username = registerModel.Username,
                    Password = registerModel.Password, 
                    EmailAddress = registerModel.EmailAddress
                };

                _context.Users.Add(user);
                await _context.SaveChangesAsync();

                _logger.LogInformation("Yeni kullanıcı kaydı başarıyla oluşturuldu.");
                return true;
            }
            catch (Exception ex)
            {
                _logger.LogError(ex, "Kullanıcı kaydı oluşturulurken bir hata oluştu.");
                throw;
            }
        }

        public async Task<bool> Login(LoginModel loginModel)
        {
            try
            {
                var user = await _context.Users
                    .FirstOrDefaultAsync(u => u.Username == loginModel.Username && u.Password == loginModel.Password);

                if (user != null)
                {
                    _logger.LogInformation("Giriş başarılı.");
                    return true;
                }

                _logger.LogWarning("Giriş başarısız: Kullanıcı adı veya şifre hatalı.");
                throw new InvalidOperationException("Kullanıcı adı veya şifre hatalı.");
            }
            catch (Exception ex)
            {
                _logger.LogError(ex, "Giriş yapılırken bir hata oluştu.");
                throw;
            }
        }
    }
}
